src/inventorymanager.cpp

```diff
@@ -278,248 +457,73 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 		return;
 	}
 
-	if (move_somewhere) {
-		s16 old_to_i = to_i;
-		u16 old_count = count;
-		caused_by_move_somewhere = true;
-		move_somewhere = false;
+	if (from_i < 0 || list_from->getSize() <= (u32) from_i) {
+		infostream << "IMoveAction::apply(): FAIL: source index out of bounds: "
+			<< "size of from_list=\"" << list_from->getSize() << "\""
+			<< ", from_index=\"" << from_i << "\"" << std::endl;
+		return;
+	}
 
+	// Do not handle rollback if both inventories are that of the same player
+	bool ignore_rollback = from_inv.type == InventoryLocation::PLAYER && from_inv == to_inv;
```

Comment:
I don't think rollback can work for detached inventories either

Reply:
It probably doesn't, but it was like that before, and I don't think it's worth changing in this PR.